scripts/views.py

```diff
@@ -424,6 +426,32 @@ def form_valid(self, form):
         return super().form_valid(form)
 
 
+class ScriptDeleteView(LoginRequiredMixin, generic.edit.BaseDeleteView):
+    """
+    Deletes a script version.
+    """
+
+    model = models.ScriptVersion
+
+    def delete(self, request, *args, **kwargs):
+        self.object: models.ScriptVersion = self.get_object()
+        script: models.Script = self.object.script
+        if script.owner != request.user:
+            raise Http404("Cannot delete a script you don't own.")
```

Comment:
This might make more sense as a 401 if the user is not logged in or a 403 in the case the user is logged in but the script does not belong to them.

Reply:
Thanks, agree that a 403 makes sense here. Django actually has a built in mixin for users not logged in that redirects to the login page so I'm going to stick with that rather than returning a 401.